presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpiller.java

```diff
@@ -74,28 +65,39 @@ public BinaryFileSpiller(
         this.serde = requireNonNull(serde, "serde is null");
         this.executor = requireNonNull(executor, "executor is null");
         this.spillerStats = requireNonNull(spillerStats, "spillerStats is null");
-        this.localSpillContext = localSpillContext;
+        this.localSpillContext = requireNonNull(localSpillContext, "localSpillContext can not be null");
         try {
-            this.targetDirectory = Files.createTempDirectory(spillPath, "presto-spill");
+            targetFileName = Files.createTempFile(spillPath, "spill", ".bin");
         }
         catch (IOException e) {
-            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Failed to create spill directory", e);
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Failed to create spill file", e);
         }
     }
 
     @Override
     public ListenableFuture<?> spill(Iterator<Page> pageIterator)
     {
-        checkState(previousSpill.isDone());
-        Path spillPath = getPath(spillsCount++);
+        checkNoSpillInProgress();
+        spillInProgress = executor.submit(
+                () -> writePages(pageIterator));
+        return spillInProgress;
+    }
 
-        previousSpill = executor.submit(() -> writePages(pageIterator, spillPath));
-        return previousSpill;
+    private void checkNoSpillInProgress()
+    {
+        checkState(spillInProgress.isDone(), "spill in progress");
     }
 
-    private void writePages(Iterator<Page> pageIterator, Path spillPath)
+    @Override
+    public Iterator<Page> getSpilledPages()
```

Comment:
The contract is here that `spill` method can be called multiple times, but user hast to wait until returned `Future` is completed before calling `spill` another time.
`getSpilledPages` returns iterator to all pages written when `spill` was called once or many times.
Before calling `getSpilledPages` user also have to wait until `Future` returned by last call to `spill` method is completed.

As this class is to be used single-threaded and from single client, it does not seem to me that adding an explicit `SpillHandle` result from `spill`, which then has to be passed to `getSpilledPages`, is very beneficial.
Yet one more object that needs to be passed around. And it would make implementation of `SingleStreamSpiller` implementation more complex as you would have to keep track of what pages were spilled
for which `SpillHandle`. Also you would need to call `getSpilledPages` multiple times on the client.

If you insist we may think of changing that. Or wait until it proves to be hard to work with.